addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,375 +1,465 @@
-                        wait = True
-                    if not wait:
-                        hr_eval_inter_obj.survey_req_waiting_answer(cr, uid, [int_id], context=context)
-
-                    if (not wait) and phase.mail_feature:
-                        body = phase.mail_body % {'employee_name': child.name, 'user_signature': child.user_id.signature,
-                            'eval_name': phase.survey_id.title, 'date': time.strftime('%Y-%m-%d'), 'time': time}
-                        sub = phase.email_subject
-                        if child.work_email:
-                            vals = {'state': 'outgoing',
-                                    'subject': sub,
-                                    'body_html': '<pre>%s</pre>' % body,
-                                    'email_to': child.work_email,
-                                    'email_from': evaluation.employee_id.work_email}
-                            self.pool.get('mail.mail').create(cr, uid, vals, context=context)
-
-        self.write(cr, uid, ids, {'state': 'wait'}, context=context)
-        return True
 
-    def button_final_validation(self, cr, uid, ids, context=None):
-        request_obj = self.pool.get('hr.evaluation.interview')
-        self.write(cr, uid, ids, {'state': 'progress'}, context=context)
-        for evaluation in self.browse(cr, uid, ids, context=context):
-            if evaluation.employee_id and evaluation.employee_id.parent_id and evaluation.employee_id.parent_id.user_id:
-                self.message_subscribe_users(cr, uid, [evaluation.id], user_ids=[evaluation.employee_id.parent_id.user_id.id], context=context)
-            if len(evaluation.survey_request_ids) != len(request_obj.search(cr, uid, [('evaluation_id', '=', evaluation.id), ('state', 'in', ['done', 'cancel'])], context=context)):
-                raise UserError(_("You cannot change state, because some appraisal forms have not been completed."))
-        return True
+class survey_user_input(models.Model):
+    _inherit = "survey.user_input"
 
-    def button_done(self, cr, uid, ids, context=None):
-        self.write(cr, uid, ids, {'state': 'done', 'date_close': time.strftime('%Y-%m-%d')}, context=context)
-        return True
+    evaluation_id = fields.Many2one('hr_evaluation.evaluation', string='Appraisal')
 
-    def button_cancel(self, cr, uid, ids, context=None):
-        interview_obj = self.pool.get('hr.evaluation.interview')
-        evaluation = self.browse(cr, uid, ids[0], context)
-        interview_obj.survey_req_cancel(cr, uid, [r.id for r in evaluation.survey_request_ids])
-        self.write(cr, uid, ids, {'state': 'cancel'}, context=context)
-        return True
+    @api.multi
+    def write(self, vals):
+        """ Trigger the _get_total_appraisal method when user fill up the appraisal form
+            and update kanban image gauge in employee evaluation.
+        """
+        res = super(survey_user_input, self).write(vals)
+        if vals.get('state'):
```

Comment:
Isn't it possible to put a dependency on the right field of evaluation model instead of hacking the write of surveys ?